staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go

```diff
@@ -149,24 +89,10 @@ func Unauthorized(s runtime.NegotiatedSerializer, supportsBasicAuth bool) http.H
 	})
 }
 
-// compressUsername maps all possible usernames onto a small set of categories
-// of usernames. This is done both to limit the cardinality of the
-// authorized_user_requests metric, and to avoid pushing actual usernames in the
-// metric.
-func compressUsername(username string) string {
-	switch {
-	// Known internal identities.
-	case username == "admin" ||
-		username == "client" ||
-		username == "kube_proxy" ||
-		username == "kubelet" ||
-		username == "system:serviceaccount:kube-system:default":
-		return username
-	// Probably an email address.
-	case strings.Contains(username, "@"):
-		return "email_id"
-	// Anything else (custom service accounts, custom external identities, etc.)
-	default:
-		return "other"
+func audiencesIntersect(apiAuds, responseAudiences authenticator.Audiences) bool {
```

Comment:
/cc @mikedanese 

To double check security critical change

Reply:
I agree this preserves the existing logic, but the method name is fairly misleading, since it short-circuits on empty parameters. Consider renaming to audiencesAreAcceptable() or something similar.

As a follow-up, given that (I think) we are now wrapping all audience-agnostic authenticators in ones that know how to return the API server audience in the response, I think we should probably drop the `len(responseAudiences) == 0` short-circuit.

Reply:
Renamed to audiencesAreAcceptable.

Will folllow-up with a PR to remove the `len(responseAudiences) == 0` short-circuit.